nil-docs/zkllvm/overview/key-components-tools.md

```diff
@@ -0,0 +1,83 @@
+# Key components and tools
+
+## zkLLVM Components
+
+The architecture of zkLLVM is modular and extendable. It can be divided into three main components.
+
+### Frontends
+
+Frontends parse high-level languages and generate abstract syntax trees (ASTs). Presently, zkLLVM supports C++ and Rust as frontends based on `clang` and `rustc`, respectively.
+
+:::info[zkLLVM and LLVM]
+
+In comparison to the vanilla LLVM project frontends, zkLLVM frontends generate ASTs that are compatible with the zkSNARK circuit generation process. These ASTs include additional types and metadata.
```

Comment:
Yes, technically they generate ASTs, but AST is only an intermediate product here. The result of both clang and rustc is LLVM IR (extended with our custom types and instructions).

Reply:
got it, amended!